Unhandled IOexception in Builder client

When we interact with the builder and we get an `IOException` in `RequestBody::writeTo`, the exception goes to our default handler and we print a scary stack trace.


```
2025-02-19 11:22:23.396 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for OkHttp Dispatcher
java.io.UncheckedIOException: java.net.SocketException: Broken pipe
at tech.pegasys.teku.infrastructure.ssz.sos.SszOutputStreamWriter.write(SszOutputStreamWriter.java:32) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.sos.SszWriter.write(SszWriter.java:25) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.sos.SszWriter.write(SszWriter.java:21) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.schema.impl.AbstractSszCollectionSchema.lambda$sszSerializeFixedVectorFast$1(AbstractSszCollectionSchema.java:161) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.TreeUtil.lambda$iterateLeavesData$1(TreeUtil.java:197) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.TillIndexVisitor.visit(TillIndexVisitor.java:43) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.LeafNode.iterate(LeafNode.java:89) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:98) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:98) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:98) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:98) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:98) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.BranchNode.iterate(BranchNode.java:95) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.TreeNode.iterateRange(TreeNode.java:94) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.tree.TreeUtil.iterateLeavesData(TreeUtil.java:192) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.schema.impl.AbstractSszCollectionSchema.sszSerializeFixedVectorFast(AbstractSszCollectionSchema.java:156) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.schema.impl.AbstractSszCollectionSchema.sszSerializeVector(AbstractSszCollectionSchema.java:143) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.schema.impl.AbstractSszListSchema.sszSerializeTree(AbstractSszListSchema.java:134) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.SszData.sszSerialize(SszData.java:63) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.infrastructure.ssz.SszData.sszSerialize(SszData.java:68) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
at tech.pegasys.teku.ethereum.executionclient.rest.OkHttpRestClient$2.writeTo(OkHttpRestClient.java:124) ~[teku-ethereum-executionclient-develop.jar:25.2.0+14-gfb9f670d65]
at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:62) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201) ~[okhttp-4.12.0.jar:?]
at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517) ~[okhttp-4.12.0.jar:?]
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.net.SocketException: Broken pipe
at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method) ~[?:?]
at java.base/sun.nio.ch.SocketDispatcher.write(Unknown Source) ~[?:?]
at java.base/sun.nio.ch.NioSocketImpl.tryWrite(Unknown Source) ~[?:?]
at java.base/sun.nio.ch.NioSocketImpl.implWrite(Unknown Source) ~[?:?]
at java.base/sun.nio.ch.NioSocketImpl.write(Unknown Source) ~[?:?]
at java.base/sun.nio.ch.NioSocketImpl$2.write(Unknown Source) ~[?:?]
at java.base/java.net.Socket$SocketOutputStream.write(Unknown Source) ~[?:?]
at okio.OutputStreamSink.write(JvmOkio.kt:56) ~[okio-jvm-3.6.0.jar:?]
at okio.AsyncTimeout$sink$1.write(AsyncTimeout.kt:102) ~[okio-jvm-3.6.0.jar:?]
at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:256) ~[okio-jvm-3.6.0.jar:?]
at okio.RealBufferedSink.write(RealBufferedSink.kt:147) ~[okio-jvm-3.6.0.jar:?]
at okhttp3.internal.http1.Http1ExchangeCodec$ChunkedSink.write(Http1ExchangeCodec.kt:311) ~[okhttp-4.12.0.jar:?]
at okio.ForwardingSink.write(ForwardingSink.kt:29) ~[okio-jvm-3.6.0.jar:?]
at okhttp3.internal.connection.Exchange$RequestBodySink.write(Exchange.kt:223) ~[okhttp-4.12.0.jar:?]
at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:256) ~[okio-jvm-3.6.0.jar:?]
at okio.RealBufferedSink$outputStream$1.write(RealBufferedSink.kt:118) ~[okio-jvm-3.6.0.jar:?]
at tech.pegasys.teku.infrastructure.ssz.sos.SszOutputStreamWriter.write(SszOutputStreamWriter.java:30) ~[teku-infrastructure-ssz-develop.jar:25.2.0+14-gfb9f670d65]
... 72 more
```

--- a/ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClient.java
+++ b/ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClient.java
@@ -16,6 +16,7 @@ package tech.pegasys.teku.ethereum.executionclient.rest;
 import static java.util.Objects.requireNonNull;
 
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.net.URL;
 import java.util.Map;
 import java.util.Optional;
@@ -120,8 +121,12 @@ public class OkHttpRestClient implements RestClient {
     return new RequestBody() {
 
       @Override
-      public void writeTo(final BufferedSink bufferedSink) {
+      public void writeTo(final BufferedSink bufferedSink) throws IOException {
+        try {
         requestBodyObject.sszSerialize(bufferedSink.outputStream());
+        } catch (final UncheckedIOException e) {
+          throw e.getCause();
+        }
       }
 
       @Override

--- a/ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClient.java
+++ b/ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/rest/OkHttpRestClient.java
@@ -123,7 +123,7 @@ public class OkHttpRestClient implements RestClient {
       @Override
       public void writeTo(final BufferedSink bufferedSink) throws IOException {
         try {
-        requestBodyObject.sszSerialize(bufferedSink.outputStream());
+          requestBodyObject.sszSerialize(bufferedSink.outputStream());
         } catch (final UncheckedIOException e) {
           throw e.getCause();
         }

